src/backend/mpi/swdsm.cpp

```diff
@@ -1012,7 +1016,13 @@ void self_upgrade(upgrade_type upgrade) {
 	}
 }
 
-void swdsm_argo_barrier(int n, upgrade_type upgrade) {
+/**
+ * @brief Global barrier for ArgoDSM - needs to be called by every thread in the
+ *        system that need coherent view of the memory
+ * @param n number of local thread participating
+ * @param upgrade the type of classification upgrade to perform
+ */
+void swdsm_argo_barrier(int n, argo::backend::upgrade_type upgrade) {
```

Comment:
This now contains the same documentation in two places. Is it not simpler to keep this to the header so that future changes do not have to be replicated in two files?